run-tests/src/utils.js

```diff
@@ -162,21 +165,89 @@ class TestRunner {
               clearInterval(poller);
               TestRunner._parseApiResult(content);
               this.build_status = content.status;
-              resolve();
+              resolve(content);
             }
           }
         });
       }, WATCH_INTERVAL);
     });
   }
 
+  static async _uploadResults(content) {
+    const promises = [];
+    core.info(`Uploading test report to artifacts for build id: ${content.id}`);
+    const { devices, id: buildId, framework } = content;
+    const rootDir = './reports';
+    const username = process.env[ENV_VARS.BROWSERSTACK_USERNAME].replace("-GitHubAction", "");
+    const accesskey = process.env[ENV_VARS.BROWSERSTACK_ACCESS_KEY];
+    const inputCapabilities = content.input_capabilities;
+    let reportEndpoint;
+    if (framework === FRAMEWORKS.espresso) {
+      if (inputCapabilities.cucumberOptions && inputCapabilities.cucumberOptions.plugins) {
+        reportEndpoint = URLS.REPORT.espresso_cucumber;
+      } else {
+        reportEndpoint = URLS.REPORT.espresso_junit;
+      }
+    } else if (framework === FRAMEWORKS.xcuitest) {
+      reportEndpoint = URLS.REPORT.xcuitest_resultbundle;
+    } else {
```

Comment:
Can we please add a check to upload only when enableResultBundle param is set to true?